bin/robot-factory

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+if [[ "$CURRENT_GIT_URL" == "$PROD_GIT_URL" && "$PROD_GIT_BRANCH" == "$CURRENT_GIT_BRANCH" ]];
```

Comment:
Do we want to move this check out into the `circle-ci` config? I _think_ I like it better there, but open to other opinions.

Reply:
Something like this? I'm agnostic. 
Didn't work when I tried it, that's how I got sent down the environment variable route but I think I know why that was.

```yaml
      - when:  # prod
          condition:
            and:
              - equal: [<< pipeline.project.git_url >>, << pipeline.parameters.prod_git_url >>]
              - equal: [<< pipeline.git.branch >>, << pipeline.parameters.prod_git_branch >>]
          steps:
            - build_robot: #build robot
```

Reply:
yeah something like that. If you can't get it to work I think it is fine to leave it in the script.